src/extension/prompts/node/agent/summarizedConversationHistory.tsx

```diff
@@ -206,20 +207,29 @@ class ConversationHistory extends PromptElement<SummarizedAgentHistoryProps> {
 
 		// Handle the possibility that we summarized partway through the current turn (e.g. if we accumulated many tool call rounds)
 		let summaryForCurrentTurn: string | undefined = undefined;
+		let thinkingForFirstRoundAfterSummarization: ThinkingData | undefined = undefined;
 		if (this.props.promptContext.toolCallRounds?.length) {
 			const toolCallRounds: IToolCallRound[] = [];
 			for (let i = this.props.promptContext.toolCallRounds.length - 1; i >= 0; i--) {
 				const toolCallRound = this.props.promptContext.toolCallRounds[i];
 				if (toolCallRound.summary) {
 					// This tool call round was summarized
 					summaryForCurrentTurn = toolCallRound.summary;
+					thinkingForFirstRoundAfterSummarization = toolCallRound.thinking;
 					break;
 				}
 				toolCallRounds.push(toolCallRound);
 			}
 
 			// Reverse the tool call rounds so they are in chronological order
 			toolCallRounds.reverse();
+
+			// For Anthropic models with thinking enabled, set the thinking on the first round
+			// so it gets rendered as the first thinking block after summarization
+			if ((this.props.endpoint.model.startsWith('claude') || this.props.endpoint.model.startsWith('Anthropic')) && thinkingForFirstRoundAfterSummarization && toolCallRounds.length > 0 && !toolCallRounds[0].thinking) {
```

Comment:
We still have tons of checks like this scattered around and it would be nice to be able to run them through the prompt registry too